dtc.js

```diff
@@ -161,7 +161,7 @@ function makeThese(stuff, quant) {
     }
 
     invArray.forEach(function (inventoryItem) {
-        if (stuff === inventoryItem.name && inventoryItem.quantity > 0) {
+        if (stuff === inventoryItem.name && inventoryItem.quantity > 0 && quant > 0) {
```

Comment:
Written in comment field of first commit. This fixes recursive calls to makeThese with negative quant.

eg a recursive call for aluminium bars, but inventory reduces needed aluminium bars to -10 000.
smelting aluminium bars could then be called with a quantity of -50 000. this would change your inventory to contain 50 000 more raw ore than you had.